lib/Storage/Google.php

```diff
@@ -650,6 +689,11 @@ public function test() {
 	}
 
 	public function hasUpdated($path, $time) {
+		// Changing virtual root must trigger an update
+		if ($this->root !== self::$prevRoot) {
```

Comment:
This shouldn't be needed. Please make the GDrive root part of the storage id in the `getId` function. This way whenever the root changes, the storage will have a different id which will cause it to have different oc_filecache entries. The update case as you expect it will work then as it will be like a brand new storage.

Reply:
See how it's done for SFTP: https://github.com/owncloud/core/blob/v10.0.3/apps/files_external/lib/Lib/Storage/SFTP.php#L170

Reply:
Ok, I will look into the changes!